### DISCUSSION TITLE
iptables -L does not report

### DISCUSSION THREAD

**Discussion by rkwesk** [ltsp/ltsp#170] [Q&A]
To check that NAT and ipforwarding when the server has two nics I needed to open epoptes and execute a local root terminal on a client and then try to ping a url or simply open the browser and try a site. I found that simply opening a terminal on the server and typing

     sudo iptables -L

on the server as I had done with ltsp5 doesn't show anything.

Richard

**Comment by alkisg** [comment]
Hey Richard,

maybe you weren't using just nat, but something more. AFAIK the command to list nat rules is:

```shell
# iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         
MASQUERADE  all  --  192.168.67.0/24      anywhere    
```

Btw, to show code in github with three backticks, the backtick [\`] is the key above the [tab] in the Greek keyboard layout, the same one where the tilde [\~] is.

**Comment by rkwesk** [comment]
Yes, ok, but it still returns empty, i.e. no rule is shown.

I run this on the server after ltsp has been successfully installed and is running.
 n"maybe you weren't using just nat, but something more" - no, it is run on a simple ltsp server with 2 nics and no other iptable rules have been added or zapped.

With ltsp5 if I ran that, the rules that were added by network manager show up.

```root@ltsp19sonylap64:~# iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination```

Richard

**Comment by alkisg** [comment]
Richard, I think you reported that:

- The clients can properly access the Internet,
- And **at the same time** the `sudo iptables -t nat -L` command shows nothing.

Can you check once more and verify that?
Because at that time, the output of the command in my test LTSP server does list the masquerade rule.

Network manager might be using more iptables rules than LTSP, but we can't just add more rules without a reason; AFAIK that single rule that LTSP uses, both works and is listed fine...

**Comment by rkwesk** [comment]
Yes, Alki, I need to check this again as I see with the laptop that when I use it elsewhere and the wired interface is not connected the command

     ip a

does NOT report an ip for that interface in spite of being listed in nm-connection-editor as having a manual 192.168.67.1 ip.

I agree that additional iptable rules should NOT be put unless a specific need so dictates.  Certainly in my case no other rules have been added by me.

Richard

**Comment by rkwesk** [comment]
Finally got back to this. I have setup the computer lab at the 2 GEL Nea Filadelfeia (Athens) which has had linux exclusively for several years. They have always used ltsp with two nics on the server and two switches, one for each subnet with the server as the gateway.

The install with ltsp19 went well with Ubuntu-mate 18.04 and the clients have access with the Internet. By the time it reached that stage I had about 5 minutes left so while the client was on the Internet I ran as root in a terminal on the server

     iptables -L

which was what I had originally done with ltsp5. No rules were shown. However, I had no more time to copy/paste from the terminal.

Richard

**Comment by alkisg** [comment]
Hi Richard,

as I said [above](https://github.com/ltsp/community/issues/15#issuecomment-529217764), the correct command is: `sudo iptables -t nat -L`

Without `-t nat` there, it's expected that it won't show any rules.

**Comment by rkwesk** [comment]
Hi Alki

Yes, I see. Yes, one rule is produced with

   ```sudo iptables -t nat -L```

so that solves what was a mystery for me. May I add, however, that when I was testing ltsp5 with Debian I found that

   ```sudo iptables -L```

produced (if I remember well) not one but three rules then. I am guessing here, is it possible that other iptable rules were used then to get the fat clients to access the Internet? So much hard work has been put into writing ltsp19 (it seems to me) so that even better performance and stability is achieved while the code has actually simplified some things. At least that is my impression as a mathematician - that it was always considered more "elegant" when a proof was produced that required fewer lines of code. In other words ltsp19 is far more elegant than ltsp5 imho.

Richard

**Comment by alkisg** [comment]
> ... produced (if I remember well) not one but three rules then.

I think that was because you were using network-manager instead of the sch-scripts if-up script.
I believe network-manager is using more iptables rules than sch-scripts / LTSP19.
Maybe network-manager is correct and more iptables rules are useful;
but I can't just blindly add them without first understanding the need for them (if any).
